sstables/consumer.hh

```diff
@@ -319,21 +365,34 @@ public:
             if (_pos == _read_bytes.size()) {
                 _u64 = unsigned_vint::deserialize(
                         bytes_view(reinterpret_cast<bytes::value_type*>(_read_bytes.get_write()), _read_bytes.size()));
-                if (read_bytes(data, _u64, *_read_bytes_where) == read_status::ready) {
+                if (std::visit( [&] (auto where) { return read_bytes(data, _u64, *where); }, _read_bytes_where) == read_status::ready) {
                     _prestate = prestate::NONE;
                     return read_status::ready;
                 }
             }
             break;
         }
         case prestate::READING_BYTES: {
-            auto n = std::min(_read_bytes.size() - _pos, data.size());
-            std::copy(data.begin(), data.begin() + n,
-                    _read_bytes.get_write() + _pos);
+            auto n = std::min(_read_bytes_len - _pos, data.size());
+            std::visit(overloaded_functor {
+                [&] (temporary_buffer<char>*) {
+                    std::copy(data.begin(), data.begin() + n, _read_bytes.get_write() + _pos);
```

Comment:
Not sure why you'd ever want to copy. In the previous implementation, if the cell spanned two sstable buffers, then we have to copy to linearize. But here, can't you always append fragments?